onnxruntime/core/providers/webnn/builders/model_builder.h

```diff
@@ -99,5 +103,93 @@ class ModelBuilder {
   static const IOpBuilder* GetOpBuilder(const Node& node);
 };
 
+// Create a scalar constant MLOperand of the specified value and data type.
+// Workaround for builer.constant(type, value) method since it has not been implemented now.
+// https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-constant-type-value
+// BTW, the spec is discussing if the builer.constant(type, value) should be dropped at
+// https://github.com/webmachinelearning/webnn/issues/475. Fix me according to the spec decision.
+//
+// This function enforces a mapping between the data_type and the value types:
+// - TensorProto_DataType_INT4    <-> int8_t
+// - TensorProto_DataType_UINT4   <-> int8_t
+// - TensorProto_DataType_BOOL    <-> bool
+// - TensorProto_DataType_UINT8   <-> uint8_t
+// - TensorProto_DataType_INT8    <-> int8_t
+// - TensorProto_DataType_FLOAT16 <-> float
+// - TensorProto_DataType_FLOAT   <-> float
+// - TensorProto_DataType_INT32   <-> int32_t
+// - TensorProto_DataType_INT64   <-> int64_t
+// - TensorProto_DataType_UINT32  <-> uint32_t
+// - TensorProto_DataType_UINT64  <-> uint64_t
+template <typename T>
+const emscripten::val& ModelBuilder::CreateScalarConstant(const int32_t& data_type, T value) {
+  std::string name = "webnn_scalar_constant_" + std::to_string(data_type) + std::to_string(value);
```

Comment:
The `data_type` is type of int, without an additional "_" to split it from value may cause some unexpected issue.

e.g. `data_type  = 1, value = 10` will be equal to `data_type = 11, value = 0`

```suggestion
  std::string name = "webnn_scalar_constant_" + std::to_string(data_type) + "_" + std::to_string(value);
```